### PR TITLE
Isolate environment variable theory tests.

### PR DESCRIPTION
**Pull request by JeremyKuhne**
Will hopefully help #16961.

### CODE REVIEW

File: src/System.Runtime.Extensions/tests/System/Environment.SetEnvironmentVariable.cs [hunk 1]
```diff
@@ -90,7 +90,7 @@ public void Default(EnvironmentVariableTarget target)
         [InlineData(EnvironmentVariableTarget.User)]
         public void ModifyEnvironmentVariable(EnvironmentVariableTarget target)
         {
-            const string varName = "Test_ModifyEnvironmentVariable";
+            string varName = $"Test_ModifyEnvironmentVariable ({target})";
```
Comment (danmosemsft) [inline src/System.Runtime.Extensions/tests/System/Environment.SetEnvironmentVariable.cs]: I didn't realize until now that env vars could hvae spaces in the name, but clearly they can..

**Comment by stephentoub** [comment]
The change looks fine, but can you elaborate on what the problem with the current code is that caused the failure and how this fixes it?

**Comment by ViktorHofer** [comment]
I guess you are changing the env variable name so that the EnvironmentVariableTarget gets printed in the logs?

**Comment by danmosemsft** [comment]
@dotnet-bot test innerloop windows_nt debug x86 build and test (https://github.com/dotnet/corefx/issues/18630)

**Comment by JeremyKuhne** [comment]
> can you elaborate on what the problem with the current code is that caused the failure and how this fixes it?

It _appears_ to be a timing issue with flushing registry data. We're creating and destroying the same exact value rapidly in succession- this avoids that.

**Pull request merged**